docs/examples/binary_classification_COMPAS.rst

```diff
@@ -0,0 +1,410 @@
+    unconstrained_predictor.fit(X_train, y_train)
+    
+print and plot data from training and test set as well as predictions with
+fairness-unaware classifier on both sets show only test data related plots by
+default - uncomment the following lines to see training data plots as well::
+
+    # show_proportions(X_train, sensitive_features_train, y_train,
+    #     description="original training data:", plot_row_index=1)
+    # show_proportions(X_train, sensitive_features_train,
+    #     unconstrained_predictor.predict(X_train), y_train,
+    #     description="fairness-unaware prediction on training data:",
+    #     plot_row_index=2)
+    show_proportions(X_test, sensitive_features_test, y_test,
+                    description="original test data:", plot_row_index=3)
+    show_proportions(X_test, sensitive_features_test,
+                    unconstrained_predictor.predict(X_test), y_test,
+                    description="fairness-unaware prediction on test data:",
+                    plot_row_index=4)
+    plt.show()
+
+Postprocessing the model to get a fair model
+============================================
+
+The idea behind postprocessing is to alter the output of the fairness-unaware
+model to achieve fairness. The postprocessing algorithm requires three input
+arguments:
+- the matrix of samples X
+- the vector of predictions y from the fairness-unaware model 
+- the vector of group attribute values A (in the code we refer to it as `sensitive_features`)
+
+The goal is to make the output fair with respect to constraints. The
+postprocessing algorithm uses one of
+- Demographic Parity (DP): :math:`P\ [\ h(X)=\hat{y}\ |\ A=a] = P\ [\ h(X)=\hat{y}\ ] \qquad \forall a, \hat{y}`
+- Equalized Odds (EO): :math:`P\ [\ h(X)=\hat{y}\ |\ A=a, Y=y] = P\ [\ h(X)=\hat{y}\ |\ Y=y\ ] \qquad \forall a, \hat{y}`
+
+where $h(X)$ is the prediction based on the input $X$, $\hat{y}$ and $y$ are
+labels, and $a$ is a sensitive feature value. In this example, we'd expect
+the postprocessed model with DP to be balanced between races. In this
+particular scenario it makes more sense to aim at fairness through accuracy
+like EO. EO does not make the same guarantees. Instead, it ensures parity
+between the subgroups of each race with label 1 in the training set, and
+parity between the subgroups of each race with label 0 in the training set.
+Applied to this scenario, this means that the subgroups of each race who
+reoffended in the past are equally likely to be predicted to reoffend (and
+therefore also equally likely not to). Similarly, there is parity between
+subgroups of each race without recidivism, but we have no parity between the
+groups with different training labels. In mathematical terms at the example
+of African-American and Caucasian:
+
+.. math::
+
+    P\ [\ \text{recidivism predicted}\ |\ \text{African-American, recidivism}\ ] = P\ [\ \text{recidivism predicted}\ |\ \text{Caucasian, recidivism}\ ], \text{e.g. } 0.95\\
```

Comment:
so this LaTeX math doesn't look readable anymore :-(

Reply:
I'll try to fix it. I know it supports proper math

Reply:
Math still looks broken :-( 
For example
![image](https://user-images.githubusercontent.com/10245648/77220629-5ac0a200-6aff-11ea-9b5b-e0265f86383e.png)

Reply:
That's the only thing keeping me from approving this right now. I could live with it looking like that provided there is a link to the generated notebook somewhere to make up for it (?)
Will the generated notebook live in readthedocs or how does that work?

Reply:
I've removed the `.rst` version of it, now if you build the branch, you'll have the formula there. Let me know if something's missing.

Reply:
I'll admit this looks pretty nice! It didn't work for some of the inline math, though, e.g. 
![image](https://user-images.githubusercontent.com/10245648/77600963-a4760780-6ec6-11ea-808a-79b6e5ecf85a.png)

Reply:
Math mode also doesn't respect line wrap, which is totally expected for this LaTeX style math :-( 
![image](https://user-images.githubusercontent.com/10245648/77601019-d25b4c00-6ec6-11ea-875e-57f5df26a651.png)

Reply:
... and charts have their x-labels cut off 
![image](https://user-images.githubusercontent.com/10245648/77601051-e99a3980-6ec6-11ea-86b6-4d50eafb20db.png)

Reply:
Overall I like this BETTER than the existing notebooks because you actually see the cell output as well.